src/views/ApartmentApp/components/StaffMenu/index.js

```diff
@@ -0,0 +1,174 @@
+    setApplications(applicationDetailsArray ?? []);
+    setLoading(false);
+  }, []);
+
+  useEffect(() => {
+    loadAllCurrentApplications();
+
+    // Generate string of today's date in ISO format for use in CSV filename
+    let date = new Date();
+    let dateStr = DateTime.fromJSDate(date).toISODate();
+    setDateStr(dateStr);
+  }, [userProfile, loadAllCurrentApplications]);
+
+  /**
+   * Generate arrays of objects to be converted to a CSV
+   * @param {ApplicationDetails[]} applicationDetailsArray an array of ApplicationDetails objects
+   */
+  const generateCSVData = useCallback((applicationDetailsArray) => {
+    let applicationsForCsv = [];
+    let applicantsForCsv = [];
+    let apartmentChoicesForCsv = [];
+    applicationDetailsArray.forEach((applicationDetails) => {
+      // Only add the applications that have been submitted
+      if (applicationDetails.DateSubmitted) {
+        let { Applicants, ApartmentChoices, ...filteredApplicationDetails } = applicationDetails;
+        applicationsForCsv.push(filteredApplicationDetails);
+
+        Applicants.forEach((applicant) => {
+          if (applicant.ApplicationID) {
+            applicantsForCsv.push(applicant);
+          } else {
+            applicantsForCsv.push({
+              ApplicationID: applicationDetails.ApplicationID,
+              ...applicant,
+            });
+          }
+        });
+
+        ApartmentChoices.forEach((apartmentChoice) => {
+          apartmentChoicesForCsv.push({
+            ApplicationID: apartmentChoice.ApplicationID ?? applicationDetails.ApplicationID,
+            ...apartmentChoice,
+          });
+        });
+      }
+    });
+    setApplicationJsonArray(applicationsForCsv);
+    setApplicantJsonArray(applicantsForCsv);
+    setApartmentChoiceJsonArray(apartmentChoicesForCsv);
+  }, []);
+
+  useEffect(() => {
+    generateCSVData(applications);
+  }, [applications, generateCSVData]);
+
+  if (loading) {
+    return <GordonLoader />;
+  } else {
+    return (
+      <Grid container justify="center" spacing={2}>
+        <Grid item xs={12} lg={10}>
+          <Card>
+            <CardHeader title="Download Apartment Applications" className="apartment-card-header" />
+            <CardContent>
+              <Grid container direction="row" spacing={2} padded>
+                <Grid item xs={12}>
+                  <Typography variant="body1">
+                    Use the buttons below to download a spreadsheet of the submitted applications
+                    for the current semester
+                  </Typography>
+                </Grid>
+              </Grid>
+              <Grid container justify="center" alignItems="center" spacing={2}>
+                <Grid item>
+                  <Button
+                    variant="contained"
+                    color="primary"
+                    startIcon={<GetAppIcon />}
+                    disabled={!authentication}
```

Comment:
While it certainly doesn't hurt to have these, it isn't possible for this component to display if the user is unauthenticated, since the containing view will only render it if `authentication` is true.

Reply:
I think it may be the case that something else was supposed to go in that `disabled` prop, though I don't remember

Reply:
Resolved in upcoming PR